frontend/src/lib/components/Timer.svelte

```diff
@@ -2,149 +2,123 @@
   import { onDestroy } from 'svelte';
   import { tweened } from 'svelte/motion';
   import { linear as easing } from 'svelte/easing';
-  import { fly } from 'svelte/transition';
- 
+
   export let countdown;
- 
+
   let now = Date.now();
   let end = now + countdown * 1000;
- 
+  let interval;
+  let isPaused;
+  let isResetting;
+  let isOver;
+  let isStarted = false;
+  const duration = 1000;
+
+  let offset = tweened(0, { duration, easing });
+  let rotation = tweened(0, { duration, easing });
+
   $: count = Math.round((end - now) / 1000);
   $: h = Math.floor(count / 3600);
   $: m = Math.floor((count - h * 3600) / 60);
   $: s = count - h * 3600 - m * 60;
- 
+
   function updateTimer() {
     now = Date.now();
+    const remaining = Math.max(end - now, 0);
+    const progress = (countdown - remaining / 1000) / countdown;
+
+    offset.set(progress);
+    rotation.set(progress * 360);
+
+    if (remaining === 0) {
+      clearInterval(interval);
+      isOver = true;
+    }
   }
- 
-  let interval = setInterval(updateTimer, 1000);
-  $: if (count === 0) clearInterval(interval);
- 
-  let isPaused;
-  let isResetting;
-  const duration = 1000;
- 
-  let offset = tweened(1, { duration, easing });
-  let rotation = tweened(360, { duration, easing });
- 
-  $: offset.set(Math.max(count - 1, 0) / countdown);
-  $: rotation.set((Math.max(count - 1, 0) / countdown) * 360);
- 
+
+  onDestroy(() => {
+    clearInterval(interval);
+  });
+
   function handleStart() {
     now = Date.now();
-    end = now + count * 1000;
+    if (!isStarted) {
+      end = now + (countdown - 1) * 1000;
+    } else {
+      end = now + count * 1000;
+    }
     interval = setInterval(updateTimer, 1000);
-    offset.set(Math.max(count - 1, 0) / countdown);
-    rotation.set((Math.max(count - 1, 0) / countdown) * 360);
     isPaused = false;
+    isStarted = true;
   }
- 
+
   function handlePause() {
-    offset.set(count / countdown);
-    rotation.set((count / countdown) * 360);
     clearInterval(interval);
     isPaused = true;
   }
- 
+
   function handleReset() {
```

Comment:
A little more documentation especially on this function - specifically the promise part - would be greatly appreciated